org.scala-ide.sdt.core/src/org/scalaide/extensions/autoedits/CloseBacktick.scala

```diff
@@ -0,0 +1,51 @@
+package org.scalaide.extensions
+package autoedits
+
+import org.eclipse.jface.text.IDocument
+import org.scalaide.core.lexical.ScalaPartitions
+import org.scalaide.core.text.TextChange
+import org.scalaide.util.eclipse.RegionUtils._
+
+object CloseBacktickSetting extends AutoEditSetting(
+  id = ExtensionSetting.fullyQualifiedName[CloseBacktick],
+  name = "Close backticks",
+  description = "Closes a typed opening backtick if necessary.",
+  partitions = Set(
+    IDocument.DEFAULT_CONTENT_TYPE,
+    ScalaPartitions.SCALADOC_CODE_BLOCK
+  )
+)
+
+trait CloseBacktick extends AutoEdit {
+
+  override def setting = CloseBacktickSetting
+
+  override def perform() = {
+    def ch(i: Int, c: Char) = {
+      val o = textChange.start + i
+      o >= 0 && o < document.length && document(o) == c
+    }
```

Comment:
looks like test for character `c` at position `start + i` but used locally with `-1` and c ```. So maybe easier would rename it to `isStartWithBacktick` ?